code/game/turfs/unsimulated/floor.dm

```diff
@@ -32,3 +32,182 @@
 /turf/unsimulated/floor/grass/New()
 	..()
 	icon_state = "grass[rand(1,4)]"
+
+/turf/unsimulated/floor/snow
+	..()
+	temperature = 233.15 // -40C in K
+	oxygen = MOLES_O2STANDARD*1.25 // 100kpa
+	nitrogen = MOLES_N2STANDARD*1.25 // 100kpa
+	name = "snow"
+	icon = 'icons/turf/newsnow.dmi'
+	icon_state = "snow0"
+	var/snowballs = 0
+	var/global/list/snowlayers = list()
+	var/global/list/dirtlayers = list()
+
+/turf/unsimulated/floor/snow/New()
+	..()
+	icon_state = "snow[rand(0,6)]"
+	for(var/direction in alldirs)

```

Comment:
PSH SMOOTHWALLING CODE SCHMOOTHWALLING CODE I USED IT ELSEWHERE. WHAT PROC SHOULD I USE HER INSTEAD, THEN?